dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue

```diff
@@ -140,30 +342,163 @@ export default defineComponent({
+    };
+  },
+  computed: {
+    namesOfFilesToUpload(): string[] {
+      const namesFromFilesToUpload = this.filesToUpload.map((el) => el.name.split(".")[0]);
+      const namesFromUploadedFiles = this.uploadFiles.map((el) => el.name.split(".")[0]);
+      return [...new Set([...namesFromFilesToUpload, ...namesFromUploadedFiles])];
     },
-    inputClass: {
-      "formkit-input": false,
-      "p-inputtext": true,
-      "w-full": true,
+  },
+  watch: {
+    confirmedSelectedKPIs: function (newValue: { label: string; value: string }[]) {
+      this.computedFinancialServicesTypes = newValue.map((el: { label: string; value: string }): string => {
+        return this.euTaxonomyKPIsModel.companyTypeToEligibilityKpis[
+          el.value as keyof typeof this.euTaxonomyKPIsModel.companyTypeToEligibilityKpis
+        ];
+      });
     },
-    postEuTaxonomyDataForFinancialsProcessed: false,
-    messageCount: 0,
-    formInputsModel: {},
-    postEuTaxonomyDataForFinancialsResponse: null,
-    humanizeString: humanizeString,
-  }),
+    reportingPeriod: function (newValue: Date) {
+      this.reportingPeriodYear = newValue.getFullYear();
+    },
+  },
   props: {
     companyID: {
       type: String,
     },
   },
+  mounted() {
+    const dataId = this.route.query.templateDataId;
+
+    if (typeof dataId === "string" && dataId !== "") {
+      this.editMode = true;
+      void this.loadEuData(dataId);
+    }
+
+    this.onThisPageLinks = [...this.onThisPageLinksStart];
+  },
+
   methods: {
+    /**
+     * Loads the Dataset by the provided dataId and pre-configures the form to contain the data
+     * from the dataset
+     *
+     * @param dataId the id of the dataset to load
+     */
+    async loadEuData(dataId: string): Promise<void> {
+      this.waitingForData = true;
+      const euTaxonomyDataForFinancialsControllerApi = await new ApiClientProvider(
+        assertDefined(this.getKeycloakPromise)()
+      ).getEuTaxonomyDataForFinancialsControllerApi();
+
+      const dataResponse =
+        await euTaxonomyDataForFinancialsControllerApi.getCompanyAssociatedEuTaxonomyDataForFinancials(dataId);
+      const dataResponseData = dataResponse.data;
+      if (dataResponseData.data?.fiscalYearEnd) {
+        this.fiscalYearEnd = new Date(dataResponseData.data.fiscalYearEnd);
+      }
+      if (dataResponseData.data?.referencedReports) {
+        const propertiesOfFilesAssignedToDataID = dataResponseData.data.referencedReports;
+        for (const key in propertiesOfFilesAssignedToDataID) {
+          this.uploadFiles.push({
+            name: key,
+            currency: propertiesOfFilesAssignedToDataID[key].currency,
+            isGroupLevel: propertiesOfFilesAssignedToDataID[key].isGroupLevel,
+            reference: propertiesOfFilesAssignedToDataID[key].reference,
+            reportDate: propertiesOfFilesAssignedToDataID[key].reportDate,
+            convertedReportDate: propertiesOfFilesAssignedToDataID[key].reportDate
```

Comment:
I think now I got it.   `reportDate` is a report date as string, and `convertedReportDate` is a report date as `Date`.

How about `reportDateAsDate` or `reportDateParsed` instead of `convertedReportDate`? 
That way it would be more explicit what is meant by.

Reply:
Ok.

Reply:
Note: Naming is ok now, but typing is odd. E.g. this makes no sense:

![image](https://user-images.githubusercontent.com/99255296/235636857-c6902c2e-58e7-43ac-991c-4c92f2a111ac.png)

Reply:
Was adressed by @fsoffel